### PR TITLE
Implement models_first_person_maze

### PR DESCRIPTION
**Pull request by yashrk**
https://github.com/raysan5/raylib/blob/master/examples/models/models_first_person_maze.c rewritten in OCaml.

The only expression I failed to translate directly is [L45](https://github.com/raysan5/raylib/blob/dc1cec8bab97e09656479bded3265fb420a0b073/examples/models/models_first_person_maze.c#L45):
```c
    model.materials[0].maps[MATERIAL_MAP_DIFFUSE].texture = texture;    // Set map diffuse texture
```
I just can't understand what `MATERIAL_MAP_DIFFUSE` in raylib-ocaml is.

Style and logic improvements are also welcome

### CODE REVIEW

**Comment by tjammer** [comment]
Thanks for the contribution!
I will have time to take a look later today. As for
``` c
    model.materials[0].maps[MATERIAL_MAP_DIFFUSE].texture = texture;    // Set map diffuse texture
```
This line is also used in the mesh generation example, see here https://github.com/tjammer/raylib-ocaml/blob/master/examples/models/models_mesh_generation.ml#L43. Note that albedo and diffuse refer to the same texture.

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ _build
 _opam
 .envrc
 shell.nix
+*~
```
Comment (tjammer) [inline .gitignore:7]: What are these, editor temp files?

File: examples/models/models_first_person_maze.ml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  cubicmap : Raylib.Texture2D.t;
+  model : Raylib.Model.t;
+  texture : Raylib.Texture2D.t;
+  map_pixels : Raylib.Color.t Raylib.CArray.t;
+  map_position : Raylib.Vector3.t;
+}
+
+let gui_init () =
+  let open Raylib in
+  init_window 800 600 "Reactive GUI";
```
Comment (tjammer) [inline examples/models/models_first_person_maze.ml:12]: Please use the name of the example for the window.
`"raylib [models] example - first person maze"`

File: examples/models/models_first_person_maze.ml [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  let map_image_width = Image.width map_image in
+  let map_image_height = Image.height map_image in
+  (* Convert image to texture to display (VRAM) *)
+  let cubicmap = load_texture_from_image map_image in
+  let mesh = gen_mesh_cubicmap map_image (Vector3.create 1.0 1.0 1.0) in
+  let model = load_model_from_mesh mesh in
+  (* Load map texture *)
+  let texture = load_texture "resources/cubicmap_atlas.png" in
+  (* Set map diffuse texture *)
+  set_material_texture
```
Comment (tjammer) [inline examples/models/models_first_person_maze.ml:38]: I didn't realize that this function exists. That's even better than the solution I linked above

**Review by tjammer — CHANGES_REQUESTED** [review]
Looks good to me, apart from the window name

File: .gitignore (hunk 1 above)
Comment (yashrk) [reply .gitignore:7]: Suffix for backup files for almost old traditional UNIX text editors, including Emacs, Vim and nano, AFAIK.

File: examples/models/models_first_person_maze.ml (hunk 2 above)
Comment (yashrk) [reply examples/models/models_first_person_maze.ml]: Thank you! Copy/pasted from the neighbor test project with RayGUI and React. Fixed.

**Review by tjammer — APPROVED** [review]

**Comment by tjammer** [comment]
Thanks!

**Pull request merged**